controllers/operator/deploys.go

```diff
@@ -516,6 +516,22 @@ func equalDeploys(first, second appsv1.Deployment) bool {
 		return false
 	}
 
+	if fRes.Limits.StorageEphemeral() != nil || sRes.Limits.StorageEphemeral() != nil {
+		if fRes.Limits.StorageEphemeral() == nil {
+			statusLog.Info("Resource limits ephemeral storage is removed")
+			return false
+		}
+		if sRes.Limits.StorageEphemeral() == nil {
+			statusLog.Info("Resource limits ephemeral storage added")
+			return false
+		}
+		if fmt.Sprint(fRes.Limits.StorageEphemeral().AsDec()) != fmt.Sprint(sRes.Limits.StorageEphemeral().AsDec()) {
+			statusLog.Info("Resource limits ephemeral storage not equal",
+				"first", fmt.Sprint(fRes.Requests.StorageEphemeral().AsDec()), "second", fmt.Sprint(sRes.Requests.StorageEphemeral().AsDec()))
+			return false
+		}
+	}
```

Comment:
```suggestion
	if fmt.Sprint(fRes.Limits.StorageEphemeral().AsDec()) != fmt.Sprint(sRes.Limits.StorageEphemeral().AsDec()) {
		statusLog.Info("Resource limits ephemeral storage not equal",
			"first", fmt.Sprint(fRes.Requests.StorageEphemeral().AsDec()), "second", fmt.Sprint(sRes.Requests.StorageEphemeral().AsDec()))
		return false
	}
```

I re-tested again, and looks like we don't even need the nil checks. Just comparing the values like how the other resources are compared seems to work fine. I didn't see any nil pointer errors.